File: domain/src/main/java/nam/tran/domain/IRepository.java
package nam.tran.domain;

import android.arch.lifecycle.LiveData;

import java.util.List;

import nam.tran.domain.entity.SoccerSeasonEntity;
import nam.tran.domain.entity.TeamEntity;
import nam.tran.domain.entity.state.Resource;

public interface IRepository {

    LiveData<Resource<List<SoccerSeasonEntity>>> getListSeason();

    LiveData<Resource<List<TeamEntity>>> getListTeam(int idSeason);
}

File: domain/src/main/java/nam/tran/domain/IRepository.kt
package nam.tran.domain

import android.arch.lifecycle.LiveData

import nam.tran.domain.entity.SoccerSeasonEntity
import nam.tran.domain.entity.TeamEntity
import nam.tran.domain.entity.state.Resource

interface IRepository {

    val listSeason: LiveData<Resource<List<SoccerSeasonEntity>>>

    fun getListTeam(idSeason: Int, type: Int): LiveData<Resource<List<TeamEntity>>>
}
